src/reconcile/rec_visibility.c

```diff
@@ -408,11 +408,18 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
     if (F_ISSET(r, WT_REC_VISIBILITY_ERR))
         WT_PANIC_ERR(session, EINVAL, "reconciliation error, update not visible");
 
-    /* If not trying to evict the page, we know what we'll write and we're done.
-     * FIXME-PM-1521: We need to save updates for checkpoints as it needs to write to history store
-     * as well
+    /*
+     * If the update we've selected was an on disk update restored from lookaside or the data store
+     * skip saving an update as it should always be the update at the end of the update chain, and
+     * therefore not have any updates that need to be written to the history store after it. If
+     * we're doing checkpoint reconciliation, and the update doesn't have any further updates that
+     * need to be written to the history store, skip saving the update as saving the update will
+     * cause reconciliation to think there is work that needs to be done when there might not be.
      */
-    if (!F_ISSET(r, WT_REC_EVICT))
+    if ((upd_select->upd != NULL && F_ISSET(upd_select->upd, WT_UPDATE_RESTORED_FROM_DISK)) ||
```

Comment:
In upd_select WT_UPDATE_RESTORED_FROM_DISK is always not set. So we can ignore this case.

Reply:
Thats not entirely true, test_bug004 is an example where WT_UPDATE_RESTORED_FROM_DISK will be set in __rec_upd_select, however this check is still not needed as its actually covered by the next-> == NULL check.

Reply:
Then that's weird. We shouldn't choose any version from las. I'll have a look at it later.

Reply:
What I'm suggesting is that we can simply append the ext upd Alex C constructed to the end of the tombstone. Anyway, it's a very cheap operation. If the page get kicked out of the memory, both the tombstone and the appended upd get discarded. If we keep the page in memory, that is also OK. In this way, we can get rid of your special check.

Reply:
If you append that update to the end of the tombstone you end up selecting the tombstone as the value to get written to the disk but it won't get squashed into the original update as the code that squashes it is the code that alex wrote. So I'm reasonably confused by your suggestion here. I did however remove the check as it wasn't required as it was being covered by the other check.

Reply:
No, Alex's change ensures we select the newly appended upd instead of the tombstone. Actually, we can conveniently append a birthmark to the tombstone, i think.

Reply:
Created WT-5338 for this.